### ISSUE TITLE
Linux/arm64 CI failing - Cannot find binary for platform linux with architecture arm64

### ISSUE THREAD

**Issue by GZLiew** [pact-foundation/pact-js-core#416]
_Thank you for reporting a bug! We appreciate it very much. Issues are a big input into the priorities for Pact development_

_All italic text in this template is safe to remove before submitting_

_Thanks again!_

### Software versions

_Please provide at least OS and version of pact-js_

- **OS**: _e.g. Linux
- **Node Version**: 16.18.1

### Issue Checklist

Please confirm the following:

- [x] I have upgraded to the latest
- [x] I have the read the FAQs in the Readme
- [x] I have triple checked, that there are **no unhandled promises** in my code
- [x] I have set my log level to debug and attached a log file showing the complete request/response cycle

### Expected behaviour

No errors in CI machine

### Actual behaviour

getting error

Error: Error while locating pact binary: Cannot find binary for platform 'linux' with architecture 'arm64'.
--
  | at throwError (/workdir/node_modules/@pact-foundation/pact-core/standalone/install.ts:7:9)
  | at Object.getBinaryEntry (/workdir/node_modules/@pact-foundation/pact-core/standalone/install.ts:51:9)
  | at Object.standalone (/workdir/node_modules/@pact-foundation/pact-core/src/pact-standalone.ts:40:5)
  | at Object.<anonymous> (/workdir/node_modules/@pact-foundation/pact-core/src/pact-standalone.ts:72:16)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .js] (/workdir/node_modules/ts-node/src/index.ts:1608:43)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.<anonymous> (/workdir/node_modules/@pact-foundation/pact-core/src/server.ts:2:1)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .js] (/workdir/node_modules/ts-node/src/index.ts:1608:43)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.<anonymous> (/workdir/node_modules/@pact-foundation/pact-core/src/pact.ts:2:1)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .js] (/workdir/node_modules/ts-node/src/index.ts:1608:43)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.<anonymous> (/workdir/node_modules/@pact-foundation/pact-core/src/index.ts:1:1)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .js] (/workdir/node_modules/ts-node/src/index.ts:1608:43)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.<anonymous> (/workdir/node_modules/@pact-foundation/src/httpPact/index.ts:2:1)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .js] (/workdir/node_modules/ts-node/src/index.ts:1608:43)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.<anonymous> (/workdir/node_modules/@pact-foundation/src/index.ts:25:1)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .js] (/workdir/node_modules/ts-node/src/index.ts:1608:43)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.<anonymous> (/workdir/src/create/create.test.ts:2:1)
  | at Module._compile (node:internal/modules/cjs/loader:1155:14)
  | at Module.m._compile (/workdir/node_modules/ts-node/src/index.ts:1618:23)
  | at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
  | at Object.require.extensions.<computed> [as .ts] (/workdir/node_modules/ts-node/src/index.ts:1621:12)
  | at Module.load (node:internal/modules/cjs/loader:1033:32)
  | at Function.Module._load (node:internal/modules/cjs/loader:868:12)
  | at Module.require (node:internal/modules/cjs/loader:1057:19)
  | at require (node:internal/modules/cjs/helpers:103:18)
  | at Object.exports.requireOrImport (/workdir/node_modules/mocha/lib/nodejs/esm-utils.js:49:16)
  | at async Object.exports.loadFilesAsync (/workdir/node_modules/mocha/lib/nodejs/esm-utils.js:91:20)
  | at async singleRun (/workdir/node_modules/mocha/lib/cli/run-helpers.js:125:3)
  | at async Object.exports.handler (/workdir/node_modules/mocha/lib/cli/run.js:370:5)
  | 🚨 Error: The command exited with status 1

**Comment by mefellows** [comment]
Thanks. I think this is captured elsewhere but will have to look.

See also https://github.com/pact-foundation/pact-js-core/pull/407 and #347

**Comment by alshdavid** [comment]
Running a Linux VM, I am unable to install pact on my local project due to this. 
```
➜  test-project uname -mo
aarch64 GNU/Linux

➜  test-project npm i @pact-foundation/pact-node@10.17.6
npm WARN deprecated @types/pino-std-serializers@4.0.0: This is a stub types definition. pino-std-serializers provides its own type definitions, so you do not need this installed.
npm WARN deprecated @types/pino-pretty@5.0.0: This is a stub types definition. pino-pretty provides its own type definitions, so you do not need this installed.
npm WARN deprecated libnpmconfig@1.2.1: This module is not used anymore. npm config is parsed by npm itself and by @npmcli/config
npm ERR! code 1
npm ERR! path /root/Development/atlassian/test/node_modules/@pact-foundation/pact-node
npm ERR! command failed
npm ERR! command sh -c node postinstall.js
npm ERR! /root/Development/atlassian/test/node_modules/@pact-foundation/pact-node/standalone/install.js:42
npm ERR!     throw new Error(chalk_1.default.red("Error while installing binary: " + msg));
npm ERR!     ^
npm ERR!
npm ERR! Error: Error while installing binary: Cannot find binary for platform 'linux' with architecture 'arm64'.
npm ERR!     at throwError (/root/Development/atlassian/test/node_modules/@pact-foundation/pact-node/standalone/install.js:42:11)
npm ERR!     at getBinaryEntry (/root/Development/atlassian/test/node_modules/@pact-foundation/pact-node/standalone/install.js:282:11)
npm ERR!     at setup (/root/Development/atlassian/test/node_modules/@pact-foundation/pact-node/standalone/install.js:286:17)
npm ERR!     at /root/Development/atlassian/test/node_modules/@pact-foundation/pact-node/standalone/install.js:319:12
npm ERR!     at Object.<anonymous> (/root/Development/atlassian/test/node_modules/@pact-foundation/pact-node/postinstall.js:3:1)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
npm ERR!     at node:internal/main/run_main_module:17:47

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-12-08T01_18_41_776Z-debug-0.log
```

**Comment by mefellows** [comment]
Thanks @alshdavid. I appreciate the problem, but don't have a straightforward solution.

If you are running docker for that "VM", you could add the `--platform linux/amd64` flag to emulate x86_64 but appreciate that might not be a long term solution.

I think this package may need to be split, or the (non-Ruby based) code moved from here into Pact JS. 

Given that _parts_ of this package (the parts that use the Ruby core) work in some architectures and others don't, I don't think we have much choice other than to state that this package doesn't work in a bunch of architectures, which I think is not right.

**Comment by mefellows** [comment]
Please follow this issue https://github.com/pact-foundation/pact-js/issues/1067 for progress. 

TL;DR - the CLI which is the root cause of this compatibility issue, will be removed from this project as a dependency in the next major release.

**Comment by benjaminlees** [comment]
@mefellows is there a work around for this as it is currently blocking me from using pact in my project.

**Comment by mefellows** [comment]
The plan above is to move the ruby binaries out of this project.

For now, there is a workaround listed here: https://github.com/pact-foundation/pact-js-core/pull/407

**Comment by YOU54F** [comment]
I've been making some changes to the travelling-ruby system that we use to package pact-ruby-standalone, in order to support arm64 on osx and linux, it also provides updates to Ruby 3.1.2 and windows 32 bit binaries (currently it only provides 64 bit)

PR here https://github.com/phusion/traveling-ruby/pull/132

Releases here https://github.com/YOU54F/pact-ruby-standalone/releases/tag/v3.1.2.2-alpha

I've tested them out against pact-js-core 

https://www.npmjs.com/package/@you54f/pact-core/v/13.14.0?activeTab=code

https://github.com/pact-foundation/pact-js-core/compare/master...YOU54F:pact-js-core:master

and against the old 9.x.x / pact-node branch 

https://www.npmjs.com/package/@you54f/pact-node/v/10.18.0?activeTab=code

The size on the pact-core is concerning tho, it's a chonki boi as it contains the pact ruby standalone and the ffi for every platform, regardless of the downloading platform.

I wondered how small we could make it if we could remove standalone, and ffi from the package and download on pre install with this janky script

https://www.npmjs.com/package/@you54f/pact-core?activeTab=code  499 kB vs 401mb (it currently sits at 215mb https://www.npmjs.com/package/@pact-foundation/pact-core)

- https://github.com/YOU54F/pact-js-core/blob/install_pact_ffi/preinstallffi.js (pre install with no external deps)
- https://github.com/YOU54F/pact-js-core/blob/install_pact_ffi/binding.gyp (more platform and arch specific node gyp file)

But these will suffer from all the problems users have of not being able to run pre/post install scripts, issues downloading binaries. 

I think the plan to split out the ruby binaries is properly wise, they can be packaged in a separate npm package, and the uplift for users should be quite small.

**Comment by YOU54F** [comment]
pact-js-core now supports linux arm64 for both the pact ruby standalone binaries, and the pact ffi

you can find a version with support for both in

https://www.npmjs.com/package/@pact-foundation/pact-core/v/13.15.0
